src/Microsoft.DotNet.Build.Tasks/PackageFiles/buildagainstpackages.targets

```diff
@@ -38,7 +38,7 @@
     </ResolveNuGetPackagesDependsOn>
   </PropertyGroup>
 
-  <Target Name="GenerateTestProjectJson" DependsOnTargets="ExtractProjectReferences;GatherProjectReferenceForProjectJson;DetermineProjectJsonPath;AddDependenciesToProjectJson" />
+  <Target Name="GenerateTestProjectJson" DependsOnTargets="ExtractProjectReferences;GatherProjectReferenceForProjectJson;DetermineProjectJsonPath;AddDependenciesToProjectJson" Condition="'$(UsesCommonTestProjectJson)'!='true'"/>
```

Comment:
By the way regardless of the condition value all the DependsOnTargets will still get executed, is that going to cause issues?